File: Solutions/JavaScript/student.test.js
const { describe, it, expect } = require('@jest/globals');
const { Student, StudentTracker } = require('./student');

describe('Student Class', () => {
    it('should calculate the average grade correctly', () => {
        const mikey = new Student('Mikey', [85, 90, 78]);
        expect(mikey.averageGrade).toBeCloseTo(84.33, 2);
    });

    it('should return the correct letter grade', () => {
        const mikey = new Student('Mikey', [95, 92, 98]);
        const brand = new Student('Brand', [80, 88, 76]);
        const chunk = new Student('Chunk', [75, 70, 68]);
        const mouth = new Student('Mouth', [65, 61, 58]);
        const troy = new Student('Troy', [55, 62, 59]);

        expect(mikey.letterGrade).toBe('A');
        expect(brand.letterGrade).toBe('B');
        expect(chunk.letterGrade).toBe('C');
        expect(mouth.letterGrade).toBe('D');
        expect(troy.letterGrade).toBe('F');
    });
});

describe('StudentTracker Class', () => {
    it('should add a student to the tracker', () => {
        const tracker = new StudentTracker();
        tracker.addStudent('Andy', [92, 88, 95]);
        expect(tracker.students.length).toBe(1);
    });

    it('should get students with a specific letter grade', () => {
        const tracker = new StudentTracker();
        const mikey = tracker.addStudent('Mikey', [95, 92, 98]);
        const chunk = tracker.addStudent('Chunk', [75, 70, 68]);
        const mouth = tracker.addStudent('Mouth', [65, 61, 58]);
        const brand = tracker.addStudent('Brand', [85, 82, 88]);
        const stef = tracker.addStudent('Stef', [91, 89, 93]);
        const troy = tracker.addStudent('Troy', [55, 62, 59]);

        const gradeAStudents = tracker.getStudentsWithLetterGrade('A');
        const gradeBStudents = tracker.getStudentsWithLetterGrade('B');
        const gradeCStudents = tracker.getStudentsWithLetterGrade('C');
        const gradeDStudents = tracker.getStudentsWithLetterGrade('D');
        const gradeFStudents = tracker.getStudentsWithLetterGrade('F');

        expect(gradeAStudents.length).toBe(2);
        expect(gradeAStudents).toContain(mikey);
        expect(gradeAStudents).toContain(stef);

        expect(gradeBStudents.length).toBe(1);
        expect(gradeBStudents).toContain(brand);

        expect(gradeCStudents.length).toBe(1);
        expect(gradeCStudents).toContain(chunk);

        expect(gradeDStudents.length).toBe(1);
        expect(gradeDStudents).toContain(mouth);

        expect(gradeFStudents.length).toBe(1);
        expect(gradeFStudents).toContain(troy);
    });
});

File: Solutions/C#/Student.Test/StudentTest.cs
using NUnit.Framework;
using static Student.Student;
using static Student.StudentTracker;

namespace Student.Test
{
    [TestFixture]
    public class StudentTests
    {
        [Test]
        public void CalculateAverageGradeCorrectly()
        {
            var mikey = new Student("Mikey", (new[] { 85, 90, 78 }).ToList());
            Assert.That(mikey.AverageGrade, Is.EqualTo(84.33).Within(0.01));
        }

        [Test]
        public void ReturnCorrectLetterGrade()
        {
            var mikey = new Student("Mikey", (new[] { 95, 92, 98 }).ToList());
            var brand = new Student("Brand", (new[] { 80, 88, 76 }).ToList());
            var chunk = new Student("Chunk", (new[] { 75, 70, 68 }).ToList());
            var mouth = new Student("Mouth", (new[] { 65, 61, 58 }).ToList());
            var troy = new Student("Troy", (new[] { 55, 62, 59 }).ToList());

            Assert.That(mikey.LetterGrade, Is.EqualTo("A"));
            Assert.That(brand.LetterGrade, Is.EqualTo("B"));
            Assert.That(chunk.LetterGrade, Is.EqualTo("C"));
            Assert.That(mouth.LetterGrade, Is.EqualTo("D"));
            Assert.That(troy.LetterGrade, Is.EqualTo("F"));
        }
    }

    [TestFixture]
    public class StudentTrackerTests
    {
        [Test]
        public void AddStudentToTracker()
        {
            var tracker = new StudentTracker();
            tracker.AddStudent("Andy", (new[] { 92, 88, 95 }).ToList());
            Assert.That(tracker.StudentCount, Is.EqualTo(1));
        }

        [Test]
        public void GetStudentsWithSpecificLetterGrade()
        {
            var tracker = new StudentTracker();
            var mikey = tracker.AddStudent("Mikey", (new[] { 95, 92, 98 }).ToList());
            var chunk = tracker.AddStudent("Chunk", (new[] { 75, 70, 68 }).ToList());
            var mouth = tracker.AddStudent("Mouth", (new[] { 65, 61, 58 }).ToList());
            var brand = tracker.AddStudent("Brand", (new[] { 85, 82, 88 }).ToList());
            var stef = tracker.AddStudent("Stef", (new[] { 91, 89, 93 }).ToList());
            var troy = tracker.AddStudent("Troy", (new[] { 55, 62, 59 }).ToList());

            var gradeAStudents = tracker.GetStudentsWithLetterGrade("A");
            var gradeBStudents = tracker.GetStudentsWithLetterGrade("B");
            var gradeCStudents = tracker.GetStudentsWithLetterGrade("C");
            var gradeDStudents = tracker.GetStudentsWithLetterGrade("D");
            var gradeFStudents = tracker.GetStudentsWithLetterGrade("F");

            Assert.That(gradeAStudents.Count, Is.EqualTo(2));
            Assert.That(gradeAStudents, Contains.Item(mikey));
            Assert.That(gradeAStudents, Contains.Item(stef));

            Assert.That(gradeBStudents.Count, Is.EqualTo(1));
            Assert.That(gradeBStudents, Contains.Item(brand));

            Assert.That(gradeCStudents.Count, Is.EqualTo(1));
            Assert.That(gradeCStudents, Contains.Item(chunk));

            Assert.That(gradeDStudents.Count, Is.EqualTo(1));
            Assert.That(gradeDStudents, Contains.Item(mouth));

            Assert.That(gradeFStudents.Count, Is.EqualTo(1));
            Assert.That(gradeFStudents, Contains.Item(troy));
        }
    }
}
